src/fabric.erl

```diff
@@ -515,3 +516,21 @@ kl_to_record(KeyList,RecName) ->
                     Index = lookup_index(couch_util:to_existing_atom(Key),RecName),
                     setelement(Index, Acc, Value)
                         end, Acc0, KeyList).
+
+prevent_upload_during_upgrade() ->
+    case upgrade_in_progress() of
+        true ->
+            timer:sleep(500),
+            prevent_upload_during_upgrade();
+        _ ->
+            ok
+    end.
+
+-spec upgrade_in_progress() -> boolean().
+upgrade_in_progress() ->
+    case config:get("cloudant", "upgrade_in_progress", "false") of
```

Comment:
since this fix need to be in dbnext as well it is better to have similar implementation. The current form of `config:get` will not work on dbnext. Since dbnext is strict on type of default argument. So I would change this function to be:
```
upgrade_in_progress() ->
     config:get_boolean("cloudant", "upgrade_in_progress", false).
```